pkg/controller/argocd/role.go

```diff
@@ -83,30 +85,54 @@ func (r *ReconcileArgoCD) reconcileRoles(cr *argoprojv1a1.ArgoCD) (role *v1.Role
 }
 
 // reconcileRole
-func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.Role, error) {
-	role := newRole(name, policyRules, cr)
-	if err := applyReconcilerHook(cr, role, ""); err != nil {
+func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) ([]*v1.Role, error) {
+	var roles []*v1.Role
+	namespaces := corev1.NamespaceList{}
+	listOption := client.MatchingLabels{
+		common.ArgoCDManagedNamespaceLabel: cr.Namespace,
+	}
+	if err := r.client.List(context.TODO(), &namespaces, listOption); err != nil {
 		return nil, err
 	}
-	existingRole := v1.Role{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: role.Name, Namespace: cr.Namespace}, &existingRole)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return nil, fmt.Errorf("failed to reconcile the role for the service account associated with %s : %s", name, err)
+
+	for _, namespace := range namespaces.Items {
+		role := newRole(name, policyRules, cr)
+		if err := applyReconcilerHook(cr, role, ""); err != nil {
+			return nil, err
 		}
-		if name == dexServer && isDexDisabled() {
-			return role, nil // Dex is disabled, do nothing
+		role.Namespace = namespace.Name
+		existingRole := v1.Role{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: role.Name, Namespace: role.Namespace}, &existingRole)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to reconcile the role for the service account associated with %s : %s", name, err)
+			}
+			if name == dexServer && isDexDisabled() {
+				continue // Dex is disabled, do nothing
+			}
+			controllerutil.SetControllerReference(cr, role, r.scheme)
+			if err := r.client.Create(context.TODO(), role); err != nil {
+				return nil, err
+			}
+			roles = append(roles, role)
+			continue
 		}
-		controllerutil.SetControllerReference(cr, role, r.scheme)
-		return role, r.client.Create(context.TODO(), role)
-	}
 
-	if name == dexServer && isDexDisabled() {
-		// Delete any existing Role created for Dex
-		return role, r.client.Delete(context.TODO(), role)
+		if name == dexServer && isDexDisabled() {
+			// Delete any existing Role created for Dex
+			if err := r.client.Delete(context.TODO(), role); err != nil {
+				return nil, err
+			}
+			roles = append(roles, role)
```

Comment:
Doing an `append` here does not look right.

Reply:
removed it!! Thanks, I had added this for something that I had in my mind, but that wouldn't work!!